notifier/dockerhub.go

```diff
@@ -0,0 +1,27 @@
+package notifier
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+)
+
+// NotifyDockerHub calls a DockerHub webhook to build a container after building the artifacts
+func (n *NotifyEntry) NotifyDockerHub(metadata NotifyMetaData) error {
+	if metadata.EventType != "success" {
+		return nil
+	}
+
+	resp, err := http.PostForm(n.Target, url.Values{"build": []string{"true"}})
+	defer resp.Body.Close()
```

Comment:
move this defer ander the `if err != nil`. If there is an error and resp is nil, this will panic once it unrolls.